src/vs/platform/extensions/common/extensions.ts

```diff
@@ -159,7 +159,7 @@ export interface IExtensionContributions {
 
 export type ExtensionKind = 'ui' | 'workspace' | 'web';
 export type ExtensionWorkspaceTrustRequestType = 'never' | 'onStart' | 'onDemand';
-export type ExtensionWorkspaceTrust = { request: 'never'; } | { request: 'onStart' | 'onDemand', description: string };
+export type ExtensionWorkspaceTrust = { request: 'never'; } | { request: 'onStart', description: string } | { request: 'onDemand', description: string, requiredForConfigurations?: string[] };
```

Comment:
@lszomoru @sbatten Changed workspace trust property to also take configurations that require trust.

Reply:
I think that this can be simplified to 
```
{ request: 'never' | 'onStart' } | { request: 'onDemand', description: string, requiredForConfigurations?: string[] }
```

Reply:
Another proposal to simplify further

```
export type ExtensionWorkspaceTrust = boolean | { request: 'onDemand', description: string, requiredForConfigurations?: string[] }
```

Reply:
I do prefer the explicit values but I also want to give @sbatten a chance to chime in.

Reply:
same, I think mixing boolean and object is less intuitive. Are we certain that we do not want to allow onStart extensions to provide a description? Perhaps that is not important to the user since it is just disabled and onDemand has more meaningful descriptions.